sycl/source/detail/config.hpp

```diff
@@ -318,6 +319,37 @@ template <> class SYCLConfig<SYCL_ENABLE_DEFAULT_CONTEXTS> {
   }
 };
 
+template <> class SYCLConfig<SYCL_QUEUE_THREAD_POOL_SIZE> {
+  using BaseT = SYCLConfigBase<SYCL_QUEUE_THREAD_POOL_SIZE>;
+
+public:
+  static int get() {
+    static std::once_flag Flag;
+    static const char *ValueStr = BaseT::getRawValue();
+    static int Value = 1;
+
+    std::call_once(Flag, [&] {
+      if (ValueStr)
+        try {
+          Value = std::stoi(ValueStr);
+        } catch (...) {
+          throw invalid_parameter_error(
+              "Invalid value for SYCL_QUEUE_THREAD_POOL_SIZE environment "
+              "variable",
+              PI_INVALID_VALUE);
+        }
+
+      if (Value < 1)
+        throw invalid_parameter_error(
+            "Invalid value for SYCL_QUEUE_THREAD_POOL_SIZE environment "
```

Comment:
Suggest either merge this `throw` with the previous one or change the massage to say that value of the env var is negative.

Reply:
Fixed.